utils/swift_build_support/swift_build_support/products/sourcekitlsp.py

```diff
@@ -20,8 +20,8 @@ def product_source_name(cls):
         return "sourcekit-lsp"
 
     @classmethod
-    def is_build_script_impl_product(cls):
-        return False
+    def needs_toolchain(cls):
```

Comment:
This is rather confusing name - pretty much anything compiled needs a toolchain.  It seems that this means if we need the just built toolchain?

Reply:
Thanks for the feedback on the name. I don’t think there will be a good name, that it is also brief. When (If?) all my plans for a newer Python build script happens, I was planning to implement a simple dependency system to avoid this kind of booleans.

This “needs toolchain” means that the product needs the just created (and installed) toolchain in the installation directory. I’m not talking about any external toolchain. To avoid passing many parameters, IndexStoreDB and SourceKitLSP relies on an installed Swift toolchain (with Foundation and dispatch on them).

The problem with the previous name is that it was going start lying as soon as the products get turned from `build-script-impl` into Python.

I accept any proposals. Maybe `needs_swift_toolchain` or `needs_installed_swift_toolchain` are better options?

Reply:
@compnerd some idea which name will be better according to you? Any name works for me, as long as it doesn't refer to “build script impl”, since I don’t think it is related to it (anymore).